modeling/model.go

```diff
@@ -0,0 +1,122 @@
+package modeling
+
+import (
+	"errors"
+	"fmt"
+)
+
+// slicecapacity is the number of max endpoints per slice
+type Model struct {
+	zones         []Zone
+	slices        []Endpointslice
+	sliceCapacity uint
+	lastResult    Stat
+	alg           RoutingAlgorithm
+	simulator     TrafficSimulator
+}
+
+// Create a model with zones, routing algorithm and traffic simulator
+//		Based on the algorithm, create the endpointslices
+func NewModel(zones []Zone, alg RoutingAlgorithm, sim TrafficSimulator) (*Model, error) {
+	model := &Model{
+		zones:         zones,
+		sliceCapacity: 100,
+		alg:           alg,
+		simulator:     sim,
+	}
+	if slices, err := model.alg.CreatingSlices(model.zones); err != nil {
+		panic(err)
+	} else {
+		model.slices = slices
+	}
+	return model, nil
+}
+
+func (m *Model) SetSliceCapacity(newCapacity uint) {
+	m.sliceCapacity = newCapacity
+}
+
+// Start simulation based on the zones and endpointslices
+func (m *Model) StartSimulation() error {
+	if stat, err := m.simulator.Simulate(m.zones, m.slices); err != nil {
+		panic(err)
+	} else {
+		m.lastResult = stat
+	}
+	return nil
+}
+
+// Update the algorithm
+//		create the new endpointslices based on the new algorithm
+func (m *Model) UpdateAlgorithm(alg RoutingAlgorithm) error {
+	if alg == nil {
+		return errors.New("Empty algorithm")
+	}
+	m.alg = alg
+	if slices, err := m.alg.CreatingSlices(m.zones); err != nil {
+		panic(err)
+	} else {
+		m.slices = slices
+	}
+	return nil
+}
+
+func (m *Model) PrintLastResult() error {
+	var totalPods int
+	for _, zone := range m.zones {
+		totalPods += zone.Endpoints
+	}
+	_, numberOfSlices, _ := m.GetEndpointslices()
```

Comment:
fixed, the first returned value is the endpointslicegroups, I think its better to keep it private inside the package since this filed is computed based on zones and algorithms. This method runs as a getter for the endpointslicegroups.